Notes for 1.49.0 (#711)
oskariorg/oskari-frontend: ReleaseNotes.md

# Release Notes

## 1.49.0

For a full list of changes see:
https://github.com/oskariorg/oskari-frontend/milestone/14?closed=1

### Build script

The build script has now been completely migrated to Webpack and the old tools-scripts have been removed. Changes for the new script from 1.48.0 is that builds no longer point to a single minifierAppsetup.json but a directory hosting apps that can have the json-file OR main.js file as it's a common use case to have at least one app for the geoportal and another for the embedded maps. 
The parameter for builds changed from "env.appdef=applications/application/guest/minifierAppSetup.json" in 1.48.0

>  to "env.appdef=applications/application" in 1.49.0. The build also generates a "main.js" file when run based on the minifierAppsetup.json. You can remove the json-file after running the build for the

Reviewer: ```suggestion
 to "env.appdef=applications/application" in 1.49.0. When the build is run it also generates a "main.js" file based on the minifierAppsetup.json. You can remove the json-file after running the build for the
```

 first time and start using the main.js as the main entrypoint for the app. 

>  Bundles can now be loaded "lazily" meaning "lazy" bundles are not part of oskari.min.js but can be loaded and started as usual when referenced. This can be used to minimize the JavaScript payload users have to download. Bundles that aren't used by all the users like admin bundles should be included in the app as "lazy".

Reviewer: ```suggestion
 Bundles can now be loaded "lazily" meaning "lazy" bundles are not part of oskari.min.js but can be loaded and started as usual when referenced. This can be used to minimize the initial JavaScript payload users have to download. Bundles that aren't used by all the users like admin bundles should be included in the app as "lazy".
```

 The devserver now passes additional http headers to Jetty so forwarded requests like logging in work properly with it.

 The separate css-files that were previously referenced on the HTML (forms.css, portal.css and overwritten.css) have all been included in the oskari.min.css file. Icons.css is still a separate file. jQuery is now included in oskari.min.js as well.

#### Other build script improvements
- Build script now provides a way to be used from another repository while referencing code from  oskari-frontend/oskari-frontend-contrib repositories enabling application specific repositories like nls-oskari/pti-frontend.
- Added a flag for build script to use absolute paths instead of relative
- Added a script to create the sprite image for customized/app-specific icons.

> - Polyfills from Babel.js have been added to the build for IE support

Reviewer: ```suggestion
- `babel-polyfill` has been added to the build for IE support
```

### Cleanup

Much of the app-specific functionalities and unmaintained code have been moved to a new repository: oskari-frontend-contrib. These can still be used with any oskari-applications, but they are not "officially" supported.
Now everything in the oskari-frontend repository should work with new versions.

Also removed CSS-files that had corresponding SCSS files so it's clear which one to edit.

### Thematic maps improvements
- Improvements for diagram flyout
- Selected indicators are now listed on the indicator selection flyout and selections can be removed from that listing without the need to go the data table flyout
- Performance improvements
- Flickering reduced on statistical timeseries/indicator switching
- Color animation added for timeseries
- Other improvements and error handling added
- The select component has been changed for indicator selection flyout

> - Area borders are now shown even on point visualization

Reviewer: ```suggestion
- Map region borders are now shown even on point visualization
```

### Other changes

> - Initial support for vectortiles (the new example app now uses OpenMapTiles as basemap)

Reviewer: ```suggestion
- Initial support for MVT vector tiles (the new example app now uses OpenMapTiles as basemap)
```

> - Vectortilelayers can be styled using the Oskari style definition OR Mapbox styles

Reviewer: ```suggestion
- Vector tile layers can be styled using the Oskari style definition OR Mapbox styles
```
